collator/src/mempool/impls/std_impl.rs

```diff
@@ -334,3 +340,198 @@ impl MempoolAdapter for MempoolAdapterStdImpl {
         Ok(())
     }
 }
+
+pub struct MempoolAdapterSingleNodeImpl(Arc<Inner>);
+
+struct Inner {
+    cache: Arc<Cache>,
+    config: Mutex<ConfigAdapter>,
+    node_peer_info: PeerInfo,
+
+    unprocessed_message_queue: Arc<std::sync::Mutex<VecDeque<Bytes>>>,
+    anchor_id: tokio::sync::Mutex<Cell<MempoolAnchorId>>,
+
+    once_flag: std::sync::Once,
+}
+
+impl Inner {
+    pub async fn start_process(&self) {
+        let consensus_config = {
+            let config_guard = self.config.lock().await;
+            let consensus_config = config_guard
+                .builder
+                .get_consensus_config()
+                .expect("There is no consensus config.");
+            consensus_config.clone()
+        };
+
+        let timeout = (consensus_config.broadcast_retry_millis
+            * (consensus_config.broadcast_retry_attempts as u16)) as u64;
+
+        let mut interval = tokio::time::interval(std::time::Duration::from_millis(timeout));
+
+        loop {
+            interval.tick().await;
+
+            let external_messages = {
+                let mut messages = self.unprocessed_message_queue.lock().unwrap();
+                let external_messages: Vec<Bytes> = messages.drain(..).collect();
```

Comment:
nit: You can `collect::<Vec<_>>()` here to remove duplicated vars

Reply:
Change container to InputBuffer.